pkgs/by-name/bu/bun/hooks/bun-config-hook.sh

```diff
@@ -0,0 +1,50 @@
+# shellcheck shell=bash
+
+bunConfigHook() {
+    echo "Executing bunConfigHook"
+
+    if [ -n "${bunRoot-}" ]; then
+      pushd "$bunRoot"
+    fi
+
+    if [ -z "${bunDeps-}" ]; then
+      echo "Error: 'bunDeps' must be set when using bunConfigHook."
+      exit 1
+    fi
+
```

Comment:
```suggestion

    # `bunDeps` should be a tarball
    unpackFile "$bunDeps"
    bunDepsCopy="$(stripHash "$bunDeps" | xargs realpath)"

```
We can extract the new tarball we created here, then continue as usual

Reply:
This has issues @getchoo it's trying to unpack the file however doesn't know how to unpack it.